src/engines/kotor/gui/ingame/hud.cpp

```diff
@@ -207,6 +213,138 @@ void HUD::setPartyMember2(Creature *creature) {
 	setPortrait(3, creature != 0, creature ? creature->getPortrait() : "");
 }
 
+void HUD::scaleWidgets(unsigned int wWidth, unsigned int wHeight) {
+	scaleWidgetToUpperLeftEdge(getWidget("LBL_MAPVIEW"), wWidth, wHeight);
```

Comment:
Hmm, a lot of code duplication there. And duplicating the widget names partially.

Maybe instead this big chunk, we should rather have a static array of structs with the names of all widgets, their visibility status from the start (taking over the list on lines 106ff) and their relative position.

Something like

```
enum Position {
	kPositionUpperLeft,
	kPositionMidLeft,
	...
	kPositionUnknown
};

struct KnownWidget {
	const *name;
	bool visible;
	Position position
};

static const KnownWidget kKnownWidgets[] = {
	{ "LBL_MAPBORDER", true, kPositionUpperLeft },
	{ "LBL_LEVELUP1", false, kPositionUpperLeft },
	...
};
```

Then in `HUD::HUD()`, we're iterating over kKnownWidgets, setting them invisible if necessary, and scaling them if necessary and a position is given. Any other property we need there?